controllers/remote/cluster_cache.go

```diff
@@ -51,21 +52,68 @@ const (
 
 // ClusterCacheTracker manages client caches for workload clusters.
 type ClusterCacheTracker struct {
-	log    logr.Logger
-	client client.Client
-	scheme *runtime.Scheme
+	log                   logr.Logger
+	clientDisableCacheFor []client.Object
+	client                client.Client
+	scheme                *runtime.Scheme
 
 	lock             sync.RWMutex
 	clusterAccessors map[client.ObjectKey]*clusterAccessor
 }
 
+// ClusterCacheTrackerOptions defines options to configure
+// a ClusterCacheTracker.
+type ClusterCacheTrackerOptions struct {
+	Log                   logr.Logger
+	ClientDisableCacheFor []client.Object
+}
+
+// Option is a configuration option supplied to NewClusterCacheTracker.
+type Option func(*ClusterCacheTrackerOptions)
+
+// SetLogger allows to override the default logger used by ClusterCacheTracker.
+func SetLogger(log logr.Logger) Option {
+	return func(o *ClusterCacheTrackerOptions) {
+		o.Log = log
+	}
+}
+
+// SetClientDisableCacheFor allows to set the objects that should not be
+// cached by the client.
+func SetClientDisableCacheFor(objs []client.Object) Option {
```

Comment:
What about
```suggestion
func SetClientDisableCacheFor(objs ...client.Object) Option {
```
So the callers can simply do `SetClientDisableCacheFor(Object1, Object2)` without creating an array/slice.

Reply:
FYI, I suggested above to not use functions, but instead just expose the Options struct

Reply:
@vincepri Regarding the options argument, are you suggesting to change the function signature from

```go
func NewClusterCacheTracker(manager ctrl.Manager, options ...Option) (*ClusterCacheTracker, error) {...}
```
to
```go
func NewClusterCacheTracker(manager ctrl.Manager, options ClusterCacheTrackerOptions) (*ClusterCacheTracker, error) {...}
```
 
this will make `options` a mandatory argument and I remember you suggesting to make it an optional one.

If we want to keep the `options` argument as optional we will have to do 
```go
func NewClusterCacheTracker(manager ctrl.Manager, options ...ClusterCacheTrackerOptions) (*ClusterCacheTracker, error) {...}
```
and then perform merging of structs to construct the final options and this doesn't look like a pattern followed elsewhere in the project.